Codecademy/gamut
# chore!: Remove `ButtonDeprecated`, `ButtonDeprecatedBase` and `InputStepper`

### Overview


Removes `ButtonDeprecated` and `ButtonDeprecatedBase` from being exported, they are still used within `AccordionButtonDeprecated` for now and have been moved within that folder, but will be removed when `Accordion` is fully swapped over to `Disclosure`. Also removes an unused and deprecated `InputStepper` and updates `ModalDeprecated` to not use `ButtonDeprecated`.


### PR Checklist

- [x] Related to JIRA ticket: [GM-736]

#### Testing Instructions



1. View the [modal deprecated story](https://668ee75bf3758252a8a2e803--gamut-preview.netlify.app/?path=/docs/molecules-modals-modaldeprecated--modal-deprecated)
2. Click "Open"
3. Confirm you see the new icon x-button and that it functions as expected
4. Confirm you do not see stories for Button Deprecated or Input Stepper
5. Visit https://668ee75bf3758252a8a2e803--gamut-preview.netlify.app/?path=/docs/molecules-pagination--controlled-navigation
6. Confirm you can interact with the controlled pagination and it works as expected
7. Visit https://668ee75bf3758252a8a2e803--gamut-preview.netlify.app/?path=/docs/molecules-tabs--controlled
8. Confirm you can interact with the controlled tabs and it works as expected

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/38005)  | [Monolith Env](https://pr-38005-monolith.dev-eks.codecademy.com) |
| Mono       | [Mono Link](https://github.com/codecademy-engineering/mono/pull/6946)  | [Mono LE Env](https://tayra.codecademy.com/workspaces/new?PR_ENV=le-pr-6946)   |




[GM-736]: https://codecademy.atlassian.net/browse/GM-736?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
